spec/index.bs

```diff
@@ -1191,131 +1209,176 @@ access to it before garbage collection by calling {{GPUBuffer/destroy()}}.
 Note: This allows the user agent to reclaim the GPU memory associated with the {{GPUBuffer}}
 once all previously submitted operations using it are complete.
 
-### <dfn method for=GPUBuffer>destroy()</dfn> ### {#buffer-destroy}
+### <dfn method for=GPUBuffer>destroy()</dfn> ### {#GPUBuffer-destroy}
 
 <div algorithm="GPUBuffer.destroy">
-    <!-- TODO(kangz) handle error buffers once we have a description of the error monad -->
-    1. If the {{[[state]]}} slot of |this| is [=buffer state/mapped for reading=] or [=buffer state/mapped for writing=]:
+  **Arguments:** (None)
 
-        1. Run the steps to unmap `"this"`
+  **Returns:** (None)
 
-    1. Set the {{[[state]]}} slot of |this| to [=buffer state/destroyed=]
-</div>
+  1. If the {{[[state]]}} slot of |this| is [=buffer state/mapped=] or [=buffer state/mapped at creation=]:
 
-## Buffer Usage ## {#buffer-usage}
+    1. Run the steps to unmap `"this"`
 
-<script type=idl>
-typedef [EnforceRange] unsigned long GPUBufferUsageFlags;
-interface GPUBufferUsage {
-    const GPUBufferUsageFlags MAP_READ      = 0x0001;
-    const GPUBufferUsageFlags MAP_WRITE     = 0x0002;
-    const GPUBufferUsageFlags COPY_SRC      = 0x0004;
-    const GPUBufferUsageFlags COPY_DST      = 0x0008;
-    const GPUBufferUsageFlags INDEX         = 0x0010;
-    const GPUBufferUsageFlags VERTEX        = 0x0020;
-    const GPUBufferUsageFlags UNIFORM       = 0x0040;
-    const GPUBufferUsageFlags STORAGE       = 0x0080;
-    const GPUBufferUsageFlags INDIRECT      = 0x0100;
-    const GPUBufferUsageFlags QUERY_RESOLVE = 0x0200;
-};
-</script>
+  1. Set the {{[[state]]}} slot of |this| to [=buffer state/destroyed=]
+
+  Issue: Handle error buffers once we have a description of the error monad.
+</div>
 
 ## Buffer Mapping ## {#buffer-mapping}
 
-An application can request to map a {{GPUBuffer}} to get its mapping which is
-an {{ArrayBuffer}} representing the {{GPUBuffer}}'s allocation. Mappings are
-requested asynchronously so that the user agent can ensure the GPU finished
-using the {{GPUBuffer}} before the application gets its mapping. Mappings can
-be requested for reading with {{GPUBuffer/mapReadAsync}} or writing with
-{{GPUBuffer/mapWriteAsync}}. A mapped {{GPUBuffer}} cannot be used by the GPU
-and must be unmapped using {{GPUBuffer/unmap}} before it can be used on the
-[=Queue timeline=].
+An application can request to map a {{GPUBuffer}} so that they can access its
+content via {{ArrayBuffer}}s that represent part of the {{GPUBuffer}}'s
+allocations. Mapping a {{GPUBuffer}} is requested asynchronously with
+{{GPUBuffer/mapAsync}} so that the user agent can ensure the GPU
```

Comment:
if there isn't a synchronous version, should we rename this to just `map`?

Reply:
I'm slightly in favor of keeping the `async` because it will make the difference with `map` in native APIs very clear to developers coming from this background.